Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -354,16 +355,41 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
 
         existing_pr = None
         if remote_repo.pull_requests:
+            user, _ = remote_repo.credentials(required=False)
+
             log.info("Checking if PR already exists...")
             existing_pr = cls.find_existing_pull_request(repository, remote_repo)
             log.info("PR #{} found.".format(existing_pr.number) if existing_pr else "PR not found.")
-            if existing_pr and not existing_pr.opened and not args.defaults and (args.defaults is False or Terminal.choose(
-                    "'{}' is already associated with '{}', which is closed.\nWould you like to create a new pull-request?".format(
-                        repository.branch, existing_pr,
-                    ), default='No',
-            ) == 'Yes'):
+
+            if existing_pr and user and existing_pr.author != user and (args.defaults or Terminal.choose(
+                "'{}' is owned by '{}'\nWould you like to take over this pull-request?".format(existing_pr, existing_pr.author),
+                default='No',
+            ) == 'No'):
+                if target == source_remote:
+                    sys.stderr.write("'{}' already exists on '{}', creating a pull-request would overwrite it\n".format(
+                        repository.branch, target,
+                    ))
+                    return 1
```

Comment:
How is this reachable? Can `source_remote` ever be a fork URL?

Reply:
This is reachable if you aren't using a `fork` workflow, which is true in certain BitBucket configurations `git-webkit` also supports.